ethsigner/signer/azure/src/test/java/tech/pegasys/ethsigner/signer/azure/AzureKeyVaultAuthenticatorTest.java

```diff
@@ -0,0 +1,147 @@
+import tech.pegasys.ethsigner.core.signing.TransactionSigner;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.keyvault.KeyIdentifier;
+import com.microsoft.azure.keyvault.KeyVaultClientCustom;
+import com.microsoft.azure.keyvault.models.KeyBundle;
+import com.microsoft.azure.keyvault.models.KeyItem;
+import com.microsoft.azure.keyvault.webkey.JsonWebKeyType;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.Ignore;
+import org.junit.Test;
+
+@Ignore
+public class AzureKeyVaultAuthenticatorTest {
+
+  private static final String clientId = System.getenv("ETHSIGNER_AZURE_CLIENT_ID");
+  private static final String clientSecret = System.getenv("ETHSIGNER_AZURE_CLIENT_SECRET");
+
+  private static final String validKeyVersion = "7c01fe58d68148bba5824ce418241092";
+
+  private final AzureKeyVaultAuthenticator authenticator = new AzureKeyVaultAuthenticator();
+  private final KeyVaultClientCustom client =
+      authenticator.getAuthenticatedClient(clientId, clientSecret);
+
+  @Test
+  public void ensureCanAuthenticateAndFindKeys() {
+    assertThat(client.apiVersion()).isEqualTo("7.0");
+
+    final PagedList<KeyItem> keys = client.listKeys("https://ethsignertestkey.vault.azure.net");
+    assertThat(keys.size()).isEqualTo(1);
+
+    final KeyItem keyItem = keys.get(0);
+    assertThat(keyItem.kid()).isEqualTo("https://ethsignertestkey.vault.azure.net/keys/TestKey");
+
+    KeyIdentifier kid = new KeyIdentifier(keyItem.kid());
+    assertThat(kid.name()).isEqualTo("TestKey");
+
+    final PagedList<KeyItem> keyVersions =
+        client.listKeyVersions("https://ethsignertestkey.vault.azure.net", "TestKey");
+    assertThat(keyVersions.size()).isEqualTo(1);
+
+    KeyItem keyVersion = keyVersions.get(0);
+    assertThat(keyVersion.kid())
+        .isEqualTo("https://ethsignertestkey.vault.azure.net/keys/TestKey/" + validKeyVersion);
+
+    kid = new KeyIdentifier(keyVersion.kid());
+    assertThat(kid.version()).isEqualTo(validKeyVersion);
+
+    final KeyBundle key = client.getKey(keyVersion.kid());
+    assertThat(key.key().kty()).isEqualTo(JsonWebKeyType.EC);
+    assertThat(key.key().crv().toString()).isEqualTo("SECP256K1");
+    assertThat(Hex.encodeHexString(key.key().x()))
+        .isEqualTo("09b02f8a5fddd222ade4ea4528faefc399623af3f736be3c44f03e2df22fb792");
+    assertThat(Hex.encodeHexString(key.key().y()))
+        .isEqualTo("f3931a4d9573d333ca74343305762a753388c3422a86d98b713fc91c1ea04842");
+  }
+
+  @Test
+  public void ensureCanFindKeysAndSign() {
+
+    final AzureKeyVaultTransactionSignerFactory factory =
+        new AzureKeyVaultTransactionSignerFactory("ethsignertestkey", client);
+
+    final TransactionSigner signer = factory.createSigner("TestKey", validKeyVersion);
+    assertThat(signer.getAddress()).isEqualTo("0xfe3b557e8fb62b89f4916b721be55ceb828dbd73");
+
+    byte[] data = {1, 2, 3};
+    final Signature signature = signer.sign(data);
```

Comment:
how does this check that is can sign? should there be an assert on the signature maybe?

Reply:
updated test